File: sdks/java/src/main/java/io/licensemanager/sdk/InstallationResult.java
package io.licensemanager.sdk;

import com.fasterxml.jackson.annotation.JsonProperty;

public class InstallationResult {
    @JsonProperty("autorizado")   private boolean authorized;
    @JsonProperty("idInstalacao") private String installationId;
    @JsonProperty("jaRegistrada") private boolean alreadyRegistered;

    public boolean isAuthorized()       { return authorized; }
    public String getInstallationId()   { return installationId; }
    public boolean isAlreadyRegistered(){ return alreadyRegistered; }
}

File: sdks/php/src/InstallationResult.php
<?php

declare(strict_types=1);

namespace LicenseManager\Sdk;

/**
 * Resultado do endpoint de validação de instalação.
 */
readonly class InstallationResult
{
    public function __construct(
        public bool $authorized,
        public ?string $installationId,
        public bool $alreadyRegistered,
    ) {}
}
